### PR TITLE
[ADD] project_template_milestone

### PR DESCRIPTION
**Pull request by patrickrwilson**
This module is an addition to the project_template and project_milestone modules. It allows the milestones to be copied when a project is created from a template that contains milestones.

### CODE REVIEW

**Comment by max3903** [comment]
@patrickrwilson You can remove the `models/__pycache__` directory. This should not be included in the commit as it contains compiled code.

No need to check if the `project_milestone` is installed, it will be installed as it is a dependency.

You should call `super()` somewhere in your method to avoid duplicate code.

**Comment by patrickrwilson** [comment]
I'll work on @max3903 suggestions, might take a bit but i'll let you all know when it's ready for more reviewing.

File: project_template_milestone/tests/test_project_template.py [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+        # Verify that the project was created successfully
+        self.assertEqual(len(new_project), 1)
+
+        # Verify that the Milestones were created successfully
+        new_milestones = self.env['project.milestone'].search([(
+            'project_id', '=', self.test_project.id)])
+        self.assertEqual(len(new_milestones), 2)
+
+        # Verify that the tasks were created successfully with milestones
+        new_tasks_milestone1 = self.env['project.task'].search([(
```
Comment (max3903) [inline project_template_milestone/tests/test_project_template.py]:
```suggestion
        self.env['project.task'].search([(
```

File: project_template_milestone/tests/test_project_template.py [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+            'milestone_id', '=', self.test_milestone_1.id)])
+        self.assertEqual(len(new_project), 2)
+
+        new_tasks_milestone2 = self.env['project.task'].search([(
```
Comment (max3903) [inline project_template_milestone/tests/test_project_template.py]:
```suggestion
        self.env['project.task'].search([(
```

**Review by max3903 — APPROVED** [review]

**Comment by patrickrwilson** [comment]
Green! Ready for more reviews. Thanks

**Review by osi-scampbell — APPROVED** [review]

**Pull request merged**